corehq/apps/es/filters.py

```diff
@@ -154,11 +154,74 @@ def geo_bounding_box(field, top_left, bottom_right):
 
     top_left_geo = GeoPoint.from_string(top_left, flexible=True)
     bottom_right_geo = GeoPoint.from_string(bottom_right, flexible=True)
+    shape = {
+        'type': 'envelope',
+        'coordinates': [
+            [float(top_left_geo.longitude), float(top_left_geo.latitude)],
+            [float(bottom_right_geo.longitude), float(bottom_right_geo.latitude)]
+        ]
+    }
+    return geo_shape(field, shape, relation='within')
+
+
+class ElasticsearchGeoJSONGeometry(TypedDict):
+    """
+    A Geometry object represents points, curves, and surfaces in
+    coordinate space.
+
+    A GeoJSON Geometry object of any type other than
+    "GeometryCollection" has a member with the name "coordinates". The
+    value of the "coordinates" member is an array. The structure of the
+    elements in this array is determined by the type of geometry.
+
+    More info: `The GeoJSON specification (RFC 7946) <https://datatracker.ietf.org/doc/html/rfc7946>`_
+    """
+
+    # `ElasticsearchGeoJSONGeometry` uses Elasticsearch types, and
+    # extends GeoJSON to include Elasticsearch's "envelope" type.
+    type: Literal[
+        'point',
+        'multipoint',
+        'linestring',
+        'multilinestring',
+        'polygon',
+        'multipolygon',
+        # 'geometrycollection' is not implemented
+        'envelope',  # A bounding rectangle, or envelope
+    ]
+
+    # NOTE: In GeoJSON and Elasticsearch, the coordinate order is
+    #   longitude, latitude (X, Y), not latitude, longitude (Y, X).
+    coordinates: list[Any]
+
+
+def geo_shape(
+    field: str,
+    shape: ElasticsearchGeoJSONGeometry,
+    relation: Literal[
+        'intersects',
+        'disjoint',
+        'within',
+        'contains',
+    ] = 'intersects',
+) -> dict[str, Any]:
```

Comment:
Are the type hints going to stay?

Reply:
Oops, I nearly forgot about those. No, I'll remove them.